### ISSUE TITLE
Explore: S Narrator focus is not in sync with keyboard, while navigating among the options in ‘Select a data source’ dropdown.

### ISSUE THREAD

**Issue by msftedad** [grafana/grafana#86652]
### What happened?

On pressing down arrow from the ‘select a data source’ dropdown, keyboard focus is moving among the options, but screen reader focus is staying on the ‘select a data source’ dropdown only. 

https://github.com/grafana/grafana/assets/93735775/e4ccf0f0-59ea-4bc9-9f4f-58ba0dfc8d05

### What did you expect to happen?

Narrator focus should be in sync with keyboard, while navigating among the options in ‘Select a data source’ dropdown. 

### Did this work before?

No

### How do we reproduce it?

1. Open the URL[Grafana Play Home - Dashboards - Grafana](https://play.grafana.org/d/000000012/grafana-play-home?orgId=1) in edge Dev browser
2. ‘Grafana Play Home’ page will open. 
3. Navigate to ‘Open menu’ button & activate it. 
4. Navigate to ‘Explore’ link & activate it. 
5. ‘Explore’ page will open. 
6. Turn on narrator using ‘Ctrl+Windows+Enter’ keys. 
7. Navigate to ‘Select a data source’ dropdown & expand it using ‘Alt+Down arrow’ key. 
8. Navigate among the options using ‘Up/Down’ arrow keys. 
9. Observe the issue. 

### Environment (with versions)?

Grafana: v11.1.0-69372 (51da96d94e)
OS: 24H2 (OS Build 26100.1) 
Browser: Edge version 125.0.2506.0 (Official build) dev (64-bit)

### Grafana platform?

None

**Comment by tskarhed** [comment]
A wild guess is that this is an issue with the `Select` component, which acts weirdly with screen reader users. This probably applies to other inputs too.

**Comment by tolzhabayev** [comment]
You can see the same thing is happening to any input that is using select component https://developers.grafana.com/ui/latest/index.html?path=/story/forms-select--basic

So ideally it should be fixed in the select component if we aim to fix it.

**Comment by msftedad** [comment]
Closing this issue as we have closed our bug as Fixed. Thanks
